src/analysis/complex/upper_half_plane.lean

```diff
@@ -0,0 +1,172 @@
+/-
+Copyright (c) 2021 Alex Kontorovich and Heather Macbeth and Marc Masdeu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex Kontorovich, Heather Macbeth, Marc Masdeu
+-/
+
+import linear_algebra.special_linear_group
+import analysis.complex.basic
+import group_theory.group_action.defs
+
+/-!
+# The upper half plane and its automorphisms
+
+This file defines `upper_half_plane` to be the upper half plane in `ℂ`.
+
+We furthermore equip it with the structure of an `SL(2,ℝ)` action by
+fractional linear transformations.
+-/
+
+noncomputable theory
+
+open matrix matrix.special_linear_group
+
+open_locale classical big_operators
+
+local attribute [instance] fintype.card_fin_even
+
+@[ext] structure upper_half_plane :=
+(point : ℂ)
+(im_pos' : 0 < point.im)
+
+localized "notation `ℍ` := upper_half_plane" in upper_half_plane
+
+namespace upper_half_plane
+
+instance : has_coe ℍ ℂ :=
+{ coe := λ z, z.point }
+
+instance : topological_space ℍ := uniform_space.to_topological_space.induced (coe : ℍ → ℂ)
+
+def im (z : ℍ) := z.point.im
+
+def re (z : ℍ) := z.point.re
+
+@[simp] lemma coe_point (z : ℍ) : z.point = (z:ℂ) := rfl
+
+@[simp] lemma coe_im (z : ℍ) : (z:ℂ).im = z.im := rfl
+
+@[simp] lemma coe_re (z : ℍ) : (z:ℂ).re = z.re := rfl
+
+lemma im_pos (z : ℍ) : 0 < z.im := z.im_pos'
+
+lemma im_nonzero (z : ℍ) : z.im ≠ 0 := z.im_pos.ne'
+
+lemma nonzero (z : ℍ) : (z:ℂ) ≠ 0 :=
+begin
+  intros h,
+  apply z.im_nonzero,
+  simp [← complex.zero_im, ← h],
+end
+
+lemma norm_sq_pos (z: ℍ) : 0 < complex.norm_sq (z:ℂ) :=
```

Comment:
There are a few more places lower down the file where spaces around the colon are missing.
```suggestion
lemma norm_sq_pos (z : ℍ) : 0 < complex.norm_sq (z : ℂ) :=
```